subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java

```diff
@@ -65,15 +63,15 @@
 public class DefaultPluginRequestApplicator implements PluginRequestApplicator {
     private final PluginRegistry pluginRegistry;
     private final PluginResolverFactory pluginResolverFactory;
-    private final PluginRepositoryRegistry pluginRepositoryRegistry;
+    private final Factory<List<ArtifactRepository>> pluginRepositoriesFactory;
```

Comment:
❌ A factory seems like the wrong abstraction here. The word "Factory" implies I get something new on each call.